### PR TITLE
Allow to use redis instead of AMQP

### PR DESCRIPTION
**Pull request by bpaquet**
At the moment, Argos is not using any features related to AMQP, so we can replace it by redis rpush / lpop.
This PR allows to do this just by changing the CLOUDAMP_URL. If we want to definitely replace AMQP per redis, there are multiple others simplifications we can do.

### CODE REVIEW

File: apps/job-core/src/amqp.js [hunk 1]
```diff
@@ -14,8 +15,13 @@ async function connect() {
 let channel
 export async function getAmqpChannel() {
   if (!channel) {
-    const connection = await connect()
-    channel = await connection.createChannel()
+    if (config.get('amqp.url').startsWith('redis://')) {
+      channel = buildAMQPToRedis(config.get('amqp.url'))
```
Comment (Izzette) [inline apps/job-core/src/amqp.js]: Does it really make sense to re-use this configuration parameter?  What about `mq.type` (default `ampq`) and `redis.url`?  Are there technical challenges to doing this?

File: apps/job-core/src/amqp_test.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    await sleep(1000)
+    channel.nack(msg, false, false)
+    counter += 1
+    if (counter == 3) {
+      await channel.close()
+    }
+  })
+}
+
+run()
```
Comment (Izzette) [inline apps/job-core/src/amqp_test.js]: Should this be a different PR?

File: apps/job-core/src/amqp_to_redis.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  queueName(queue) {
+    return 'queue_' + queue
+  }
+
+  ack(message) {
+    const callback = this.ack_callbacks[message.id]
+    if (callback) {
+      this.ack_callbacks[message.id] = null
```
Comment (Izzette) [inline apps/job-core/src/amqp_to_redis.js]: Does this every get `delete`-ed?  Or will this array just fill up with nulls until it crashes from OOM?

Comment (Izzette) [reply apps/job-core/src/amqp_to_redis.js]: Can we simply:
```suggestion
      delete this.ack_callbacks[message.id]
```

File: apps/job-core/src/amqp_to_redis.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      callback()
```
Comment (Izzette) [inline apps/job-core/src/amqp_to_redis.js]: Should this be done synchronously?  This will block other messages from being received or processed as the callback is running.  Is this desirable behaviour or am I missing something?

File: apps/job-core/src/amqp_to_redis.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    }
+  }
+
+  nack(message, _allUpTo = false, _requeue = false) {
```
Comment (Izzette) [inline apps/job-core/src/amqp_to_redis.js]: Are `_allUpTo` or `_requeue` used externally?

File: apps/job-core/src/amqp_to_redis.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    this.ack(message)
+  }
+
+  sendToQueue(queue, message, _opts) {
```
Comment (Izzette) [inline apps/job-core/src/amqp_to_redis.js]: Here too, is `_opts` used externally?

File: apps/job-core/src/amqp.js (hunk 1 above)
Comment (bpaquet) [reply apps/job-core/src/amqp.js]: Waiting a decision from @gregberge. If we fully move t Redis, tes we will improve the config

File: apps/job-core/src/amqp_test.js (hunk 2 above)
Comment (bpaquet) [reply apps/job-core/src/amqp_test.js]: Good question. Waiting a review

File: apps/job-core/src/amqp_to_redis.js (hunk 3 above)
Comment (bpaquet) [reply apps/job-core/src/amqp_to_redis.js]: No risk of OOM, because the app is written for AMQP

File: apps/job-core/src/amqp_to_redis.js (hunk 4 above)
Comment (bpaquet) [reply apps/job-core/src/amqp_to_redis.js]: Nope, we are duplicating the behaviour of the AMQP API.

File: apps/job-core/src/amqp_to_redis.js (hunk 5 above)
Comment (bpaquet) [reply apps/job-core/src/amqp_to_redis.js]: AMQP API :)

File: apps/job-core/src/amqp_to_redis.js (hunk 6 above)
Comment (bpaquet) [reply apps/job-core/src/amqp_to_redis.js]: AMQP API :)

File: apps/job-core/src/amqp_to_redis.js (hunk 5 above)
Comment (Izzette) [reply apps/job-core/src/amqp_to_redis.js]: Generally parameters starting with `_` are private, so I'm not quite sure they're used as part of the AMQP API.  Moreover, in javascript, you don't need to declare this parameters in order to permit other code to pass them.  Since you're not used, you can just delete them with no side-effects.

```
 0 ✓ isabelle.cowan-bergman@mbp-de-isabelle.home oauth2_proxy $ node
Welcome to Node.js v12.13.0.
Type ".help" for more information.
> function foo(blah) {
... console.log(blah)
... }
undefined
> foo("blah")
blah
undefined
> foo("blah", "extra")
blah
undefined
> foo("blah", extra => "extra")
blah
undefined
```

File: apps/job-core/src/amqp_to_redis.js (hunk 3 above)
Comment (Izzette) [reply apps/job-core/src/amqp_to_redis.js]: I'm not sure I understand.  Maybe the maintainers will be able to shed some light.

File: apps/job-core/src/amqp_to_redis.js (hunk 4 above)
Comment (Izzette) [reply apps/job-core/src/amqp_to_redis.js]: That's all I need to hear 😊

File: apps/job-core/src/amqp_to_redis.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+      else {
+        await this.exec_message(res, message_callback)
+      }
+    }
+  }
+}
+
+export const buildAMQPToRedis = (redis_url) => {
+  return new AMQPToRedis(redis_url)
+}
```
Comment (Izzette) [inline apps/job-core/src/amqp_to_redis.js]: Is the no-trailing-newline thing in all the files for this project?
